library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPOJOOutputOperator.java

```diff
@@ -63,22 +58,18 @@
  * @since 2.1.0
  */
 @org.apache.hadoop.classification.InterfaceStability.Evolving
-public class JdbcPOJOOutputOperator extends AbstractJdbcTransactionableOutputOperator<Object>
-    implements Operator.ActivationListener<OperatorContext>
+public abstract class AbstractJdbcPOJOOutputOperator extends AbstractJdbcTransactionableOutputOperator<Object>
 {
-  @NotNull
-  private List<FieldInfo> fieldInfos;
+  private List<JdbcFieldInfo> fieldInfos;
```

Comment:
Yes, fieldInfos can be null if the statement is an insert statement. In that case, it is generated by the operator code. I have inserted checks for this in the concrete POJO output operator.